docs/advanced.rst

```diff
@@ -505,3 +505,180 @@ Besides the stage directory, additional mount points can be specified through th
+By default, the :func:`depends_on` function will create dependencies between the individual test cases of the :class:`OSULatencyTest` and the :class:`OSUBuildTest`, such that the :class:`OSULatencyTest` using ``PrgEnv-gnu`` will depend on the outcome of the :class:`OSUBuildTest` using ``PrgEnv-gnu``, but not on the outcome of the :class:`OSUBuildTest` using ``PrgEnv-intel``.
+This behaviour can be changed, but we will return to this later.
+You can create arbitrary test dependency graphs, but they need to be acyclic.
+If ReFrame detects cyclic dependencies, it will refuse to execute the set of tests and will issue an error pointing out the cycle.
+
+A ReFrame test with dependencies will execute, i.e., enter its `setup` stage, only after `all` of its dependencies have succeeded.
+If any of its dependencies have failed, the current test will be marked as failure as well.
+
+The next step for the :class:`OSULatencyTest` is to set its executable to point to the binary produced by the :class:`OSUBuildTest`.
+This is achieved with the following specially decorated function:
+
+.. literalinclude:: ../tutorial/advanced/osu/osu_benchmarks.py
+   :lines: 32-38
+
+The ``@require_deps`` decorator will bind the arguments passed to the decorated function to the result of the dependency that each argument names.
+In this case, it binds the ``OSUBuildTest`` function argument to the result of a dependency named ``OSUBuildTest``.
+In order for the binding to work correctly the function arguments must be named after the target dependencies.
+However, referring to a dependency only by the test's name is not enough, since a test might be associated with multiple programming environments.
+For this reason, a dependency argument is actually bound to a function that accepts as argument the name of a target programming environment.
+If no arguments are passed to that function, as in this example, the current programming environment is implied, such that ``OSUBuildTest()`` is equivalent to ``OSUBuildTest(self.current_environ.name)``.
+This call returns the actual test case of the dependency that has been executed.
+This allows you to access any attribute from the target test, as we do in this example by accessing the target test's stage directory, which we use to construct the path the executable.
+This concludes the presentation of the :class:`OSULatencyTest` test. The :class:`OSUBandwidthTest` is completely analogous.
+
+The :class:`OSUAllreduceTest` shown below is similar to the other two, except that it is parameterized.
+It is essentially a scalability test that is running the ``osu_allreduce`` executable created by the :class:`OSUBuildTest` for 2, 4, 8 and 16 nodes.
+
+.. literalinclude:: ../tutorial/advanced/osu/osu_benchmarks.py
+   :lines: 64-84
+
+The full set of OSU example tests is shown below:
+
+.. literalinclude:: ../tutorial/advanced/osu/osu_benchmarks.py
+
+Notice that the order dependencies are defined in a test file is irrelevant.
```

Comment:
```suggestion
Notice that the order in which dependencies are defined in a test file is irrelevant.
```

Reply:
I think mine is correct, too. A bit less formal, but the whole tutorial is informal.